packages/js/product-editor/src/hooks/use-product-type.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import {
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+
+export function useProductType() {
+	const currentSettings = useSelect( ( select ) => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		return select( blockEditorStore ).getSettings();
+	}, [] );
+
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore This action exists in the block editor store.
+	const { updateSettings } = useDispatch( blockEditorStore );
+
+	function setProductType( productType: string ) {
+		const template = productBlockEditorSettings.templates[ productType ];
+		if ( ! template ) {
+			// eslint-disable-next-line no-console
+			console.warn( 'Product type template not found: ', productType );
+			return;
+		}
+		updateSettings( {
+			...currentSettings,
+			template,
+		} );
```

Comment:
So, the idea is a single product template per product type? If so, that answers some of my questions in the general comments and helps frame what we are thinking about.

I'm wonder how extensions will add their changes to the templates... would they have to explicitly add a field to each product type template? I don't necessarily see that as a bad thing. I like that everything is explicitly and clearly opt-in. It makes the 3PD think about how their extension would work in each product type scenario.